app/code/Magento/Downloadable/Observer/IsAllowedGuestCheckoutObserver.php

```diff
@@ -15,20 +15,34 @@ class IsAllowedGuestCheckoutObserver implements ObserverInterface
      */
     const XML_PATH_DISABLE_GUEST_CHECKOUT = 'catalog/downloadable/disable_guest_checkout';
 
+    /**
+     *  Xml path to get downloadable Shareable setting
+     */
+    const XML_PATH_DOWNLOADABLE_SHAREABLE = 'catalog/downloadable/shareable';
+
     /**
      * Core store config
      *
      * @var \Magento\Framework\App\Config\ScopeConfigInterface
      */
     protected $_scopeConfig;
 
+    /**
+     * Downloadable link collection factory
+     *
+     * @var \Magento\Downloadable\Model\ResourceModel\Link\CollectionFactory
+     */
+    protected $_linksFactory;
+
     /**
      * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
      */
     public function __construct(
-        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
+        \Magento\Downloadable\Model\ResourceModel\Link\CollectionFactory $linksFactory
```

Comment:
Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)